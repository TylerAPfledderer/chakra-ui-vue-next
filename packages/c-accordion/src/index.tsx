export {
  CAccordion,
  CAccordionButton,
  CAccordionIcon,
  CAccordionItem,
  CAccordionPanel,
  ExpandedValues,
} from "./accordion"

export type {
  CAccordionButtonProps,
  CAccordionContext,
  CAccordionIconProps,
  CAccordionItemContext,
  CAccordionItemProps,
  CAccordionPanelProps,
  CAccordionProps,
} from "./accordion"
