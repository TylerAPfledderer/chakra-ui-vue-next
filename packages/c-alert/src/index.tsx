export {
  AlertVariant,
  CAlert,
  CAlertDescription,
  CAlertIcon,
  CAlertTitle,
} from "./alert"
