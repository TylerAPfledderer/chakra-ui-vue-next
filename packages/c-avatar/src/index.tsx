export * from "./c-avatar"
export * from "./c-avatar-badge"