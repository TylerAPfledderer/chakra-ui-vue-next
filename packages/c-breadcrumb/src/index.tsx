export {
  CBreadcrumb,
  CBreadcrumbItem,
  CBreadcrumbLink,
  CBreadcrumbSeparator,
} from "./c-breadcrumb"

export type {
  BreadcrumbItemProps,
  BreadcrumbLinkProps,
  BreadcrumbOptions,
  BreadcrumbProps,
  BreadcrumbSeparatorProps,
} from "./c-breadcrumb"
