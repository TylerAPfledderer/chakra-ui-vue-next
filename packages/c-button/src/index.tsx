export type { ButtonGroupProps } from "./button-group"
export { CButton } from "./button"
export { CButtonGroup } from "./button-group"
export type { CButtonProps } from "./button"
export type { CButtonSpinnerProps } from "./button"
export { CIconButton } from "./icon-button"
export type { CIconButtonProps } from "./icon-button"
export { useButtonGroup } from "./button-group"
