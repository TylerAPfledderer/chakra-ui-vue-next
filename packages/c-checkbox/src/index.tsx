export { CCheckbox } from "./checkbox"
export {
  CCheckboxGroup,
  CheckboxGroupContext,
  useCheckboxGroupContext,
} from "./checkbox-group"

export type { CCheckboxControlProps, CCheckboxProps } from "./checkbox"
export type { CCheckboxGroupProps } from "./checkbox-group"
