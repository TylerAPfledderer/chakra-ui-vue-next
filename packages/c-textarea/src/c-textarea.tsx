/**
 * Hey! Welcome to @chakra-ui/vue-next CTextarea
 *
 * C textarea
 *
 * @see Docs     https://next.vue.chakra-ui.com/c-textarea
 * @see Source   https://github.com/chakra-ui/chakra-ui-vue-next/blob/master/packages/c-textarea/src/c-textarea/c-textarea.ts
 * @see WAI-ARIA https://www.w3.org/TR/wai-aria-practices-1.2
 */

import {
  h,
  defineComponent,
  PropType,
  computed,
  reactive,
  toRefs,
  ToRefs,
} from "vue"
import {
  chakra,
  HTMLChakraProps,
  omitThemingProps,
  ThemingProps,
  useStyleConfig,
} from "@chakra-ui/vue-system"
import { omit } from "@chakra-ui/utils"
import { vueThemingProps } from "@chakra-ui/vue-utils"
import {
  FormControlOptions,
  formControlProps,
  useFormControl,
} from "@chakra-ui/c-form-control"

interface TextareaOptions {
  /**
   * The border color when the textarea is focused. Use color keys in `theme.colors`
   * @example
   * focusBorderColor = "blue.500"
   */
  focusBorderColor?: string
  /**
   * The border color when the textarea is invalid. Use color keys in `theme.colors`
   * @example
   * errorBorderColor = "red.500"
   */
  errorBorderColor?: string
}

type Omitted = "disabled" | "required" | "readOnly"

export interface CTextareaNativeProps
  extends TextareaOptions,
    FormControlOptions {}

export interface TextareaProps
  extends Omit<HTMLChakraProps<"textarea">, Omitted>,
    CTextareaNativeProps,
    ThemingProps<"Textarea"> {}

export const CTextarea = defineComponent({
  name: "CTextarea",
  props: {
    modelValue: String as PropType<string>,
    ...formControlProps,
    rows: Number as PropType<HTMLTextAreaElement["rows"]>,
    ...vueThemingProps,
  },
  emits: ["update:modelValue", "input", "change"],
  setup(props, { emit, attrs }) {
    const styles = useStyleConfig("Textarea", props)

    const ownProps = computed(() =>
      toRefs(reactive(omitThemingProps(props as ThemingProps<"Textarea">)))
    )

    const textarea = useFormControl(
      ownProps.value as ToRefs<CTextareaNativeProps>
    )

    const textareaStyles = computed(() =>
      props.rows
        ? omit(styles.value, ["h", "minH", "height", "minHeight"])
        : styles.value
    )

    const handleInput = (e: Event) => {
      emit("update:modelValue", (e?.currentTarget as HTMLInputElement)?.value)
      emit("input", e, (e?.currentTarget as HTMLInputElement)?.value)
      emit("change", e, (e?.currentTarget as HTMLInputElement)?.value)
    }

    return () => (
      <chakra.textarea
        __chakraIsRaw
        {...textarea.value}
        value={props.modelValue}
        onInput={handleInput}
        __css={textareaStyles.value}
        __label="textarea"
        {...attrs}
      />
    )
  },
})
