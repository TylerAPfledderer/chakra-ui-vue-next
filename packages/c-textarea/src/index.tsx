export { CTextarea } from "./c-textarea"
export type { CTextareaNativeProps, CTextareaProps } from "./c-textarea"
