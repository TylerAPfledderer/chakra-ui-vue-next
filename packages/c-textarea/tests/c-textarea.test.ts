import { render } from "../../test-utils/src"
import { CTextarea } from "../src"

it("should render properly", () => {
  const { asFragment } = render(CTextarea)
  expect(asFragment()).toMatchSnapshot()
})
