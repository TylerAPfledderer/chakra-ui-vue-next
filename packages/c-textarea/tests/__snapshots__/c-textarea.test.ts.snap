// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<DocumentFragment>
  <textarea
    class="chakra-textarea chakra-no2ijq"
  />
</DocumentFragment>
`;
